util/src/main/java/io/kubernetes/client/util/Cache.java

```diff
@@ -0,0 +1,269 @@
+package io.kubernetes.client.util;
+
+import com.squareup.okhttp.Call;
+import com.squareup.okhttp.ResponseBody;
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.ApiException;
+import io.kubernetes.client.models.V1ObjectMeta;
+import java.io.Closeable;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Cache<ApiType, ApiListType> implements Closeable {
+  private Function<CallGeneratorParams, Call> callGenerator;
+  private ApiClient client;
+  private Watch<ApiType> watch;
+  private Type watchType;
+  private ApiListType list;
+  private Class<?> listClass;
+  private Thread watchThread;
+  private Integer timeout;
+  private Map<String, List<ApiType>> namespaceIndex;
+
+  private static final Logger log = LoggerFactory.getLogger(Cache.class);
+
+  private Cache() {}
```

Comment:
delete this, I don't think it's needed.

Reply:
resolved in #461

[https://github.com/kubernetes-client/java/pull/461/files#diff-b32991df63ff2abba023a54e6c78a4b7R40](https://github.com/kubernetes-client/java/pull/461/files#diff-b32991df63ff2abba023a54e6c78a4b7R40)